src/amoc_metrics.erl

```diff
@@ -75,16 +72,53 @@ metric_report_datapoints(gauge)     -> [value];
 metric_report_datapoints(spiral)    -> [count, one];
 metric_report_datapoints(histogram) -> [mean, min, max, median, 95, 99, 999].
 
-create_metric_and_maybe_subscribe(ExName, Type, Reporter, Interval, Datapoints) ->
-    ok = exometer:re_register(ExName, Type, []),
+exometer_metric_type(counters) -> spiral;
+exometer_metric_type(times)    -> histogram;
+exometer_metric_type(gauge)    -> gauge.
+
+create_metric_and_maybe_subscribe(ExName, ExType) ->
+    ok = exometer:re_register(ExName, ExType, []),
+    Datapoints = metric_report_datapoints(ExType),
+    maybe_subscribe(ExName, Datapoints).
+
+maybe_add_reporter() ->
+    Reporter = get_reporter(),
+    case lists:keyfind(Reporter, 1, exometer_report:list_reporters()) of
+        {Reporter, _} -> ok;
+        _->
+            case amoc_config:get(graphite_host) of
+                undefined -> ok;
+                Host ->
+                    Prefix =amoc_config:get(graphite_prefix,net_adm:localhost()),
+                    Port = amoc_config:get(graphite_port,2003),
+                    Options = [{module, exometer_report_graphite},
+                               {prefix, Prefix},
+                               {host, Host},
+                               {port, Port},
+                               {api_key, ""}],
+                    exometer_report:add_reporter(Reporter,Options)
+            end
+    end.
+
+subsribe_default_metrics() ->
+    maybe_subscribe([amoc, users], [size]),
+    maybe_subscribe([erlang, system_info], [port_count, process_count]),
+    maybe_subscribe([erlang, memory], [total, processes, processes_used, system, binary, ets]),
+    maybe_subscribe([amoc, times, connection], metric_report_datapoints(histogram)),
+    maybe_subscribe([amoc, counters, connections], metric_report_datapoints(spiral)),
+    maybe_subscribe([amoc, counters, connection_failures], metric_report_datapoints(spiral)).
+
+get_reporter() ->
+    amoc_config:get(metrics_reporter, ?AMOC_DEFAULT_METRICS_REPORTER).
```

Comment:
Thanks for making the metrics reporter configurable. What will be returned here if there is `AMOC_metrics_reporter` env var set to, say, `my_own_reporter`. Will it be atom (as we expect it to be) or a string value?

Reply:
now it can be either `AMOC_metrics_reporter` or `AMOC_METRICS_REPORTER` os env. (however for the `2.0` i would prefer to restrict it to the "all capitals" option) or even `metrics_reporter` app env. configured in app.config

it must be atom, but the value of env variable is evaluated. so it should look like this `export AMOC_METRICS_REPORTER="some_atom"`

documentation for amoc_config is really good idea. it can contain API description, AMOC default options description and some reference how to pass scenario configurations properly.

Reply:
I think we need a test for that case :)

Reply:
in my example that is atom, but if you would have something like:
`export AMOC_SOME_VAR="[\"some_atom\"]"`
than it will be a list with one string element. the value of the os. env. is evaluated